src/custom_request_schedule_manager.cc

```diff
@@ -78,25 +77,74 @@ CustomRequestScheduleManager::GenerateSchedule()
 
 std::vector<RateSchedulePtr_t>
 CustomRequestScheduleManager::CreateWorkerSchedules(
-    const std::vector<float>& schedule)
+    const std::vector<std::chrono::milliseconds>& schedule)
 {
   std::vector<RateSchedulePtr_t> worker_schedules =
       CreateEmptyWorkerSchedules();
   std::vector<size_t> thread_ids{CalculateThreadIds()};
-  std::chrono::nanoseconds next_timestamp(0);
   size_t thread_id_index = 0;
   size_t worker_index = 0;
 
-  for (const float& val : schedule) {
-    next_timestamp = std::chrono::duration_cast<std::chrono::nanoseconds>(
-        std::chrono::duration<float>(val));
+  for (const auto& timestamp : schedule) {
+    const auto timestamp_ns{
+        std::chrono::duration_cast<std::chrono::nanoseconds>(timestamp)};
     worker_index = thread_ids[thread_id_index];
     thread_id_index = ++thread_id_index % thread_ids.size();
-    worker_schedules[worker_index]->intervals.emplace_back(next_timestamp);
+    worker_schedules[worker_index]->intervals.push_back(timestamp_ns);
   }
   SetScheduleDurations(worker_schedules);
 
   return worker_schedules;
 }
 
+void
+CustomRequestScheduleManager::InitManagerFinalize()
+{
+  schedule_ = GetScheduleFromDataset();
+  parser_->Inputs()->erase("timestamp");
+}
+
+std::vector<std::chrono::milliseconds>
+CustomRequestScheduleManager::GetScheduleFromDataset() const
+{
+  std::vector<std::chrono::milliseconds> schedule{};
+
+  if (data_loader_->GetDataStreamsCount() != 1) {
+    throw std::runtime_error(
+        "Expected input data JSON to have one stream. Fixed schedule mode "
+        "must have an input data JSON with a single flat array for the "
+        "\"data\" field with one element per request payload.");
+  }
+
+  const size_t dataset_size{data_loader_->GetTotalSteps(0)};
+
+  for (size_t dataset_index{0}; dataset_index < dataset_size; ++dataset_index) {
+    const auto timestamp{GetTimestamp(dataset_index)};
+    schedule.push_back(timestamp);
+  }
+
+  std::sort(schedule.begin(), schedule.end());
```

Comment:
Interestingly, the schedule doesn't _need_ to be supplied in order, so we guarantee it's in order here before divvying up among threads.